data/changelog/ppxlib/2024-07-25-ppxlib-0.33.0.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Ppxlib 0.33.0
+tags: [ppxlib, platform]
+changelog: |
+  - Fix a bug where `Code_path.main_module_name` would not properly remove
+  extensions from the filename and therefore return an invalid module name.
+  (#512, @NathanReb)
+
+  - Add `-unused-type-warnings` flag to the driver to allow users to disable
+  only the generation of warning 34 silencing structure items when using
+  `[@@deriving ...]` on type declarations. (#511, @mbarbin, @NathanReb)
+
+  - Make `-unused-code-warnings` flag to the driver also controls the generation
+  of warning 34 silencing structure items when using `[@@deriving ...]` on type
+  declarations. (#510, @mbarbin, @NathanReb)
+
+  - Driver: Add `-unused-code-warnings=force` command-line flag argument. (#490, @mbarbin)
+
+  - New functions `Ast_builder.{e,p}list_tail` that take an extra tail
+  expression/pattern argument parameter compared to `Ast_builder.{e,p}list`, so
+  they can build ASTs like `a :: b :: c` instead of only `[ a; b ]`.
+  (#498, #502, @v-gb, @NathanReb)
+
+  - Fix `Longident.parse` so it also handles indexing operators such as
+  `.!()`, `.%(;..)<-`, or `Vec.(.%())` (#494, @octachron)
+
+  - Add a `special_function'` variant which directly takes a `Longident.t`
+  argument to avoid the issue that `Longident.t` cover distinct syntaxic classes
+  which cannot be easily parsed by a common parser (#496, @octachron).
```

Comment:
```suggestion
  - Add a `special_function'` variant that directly takes a `Longident.t`
  argument in order to avoid the issue where `Longident.t` covers distinct syntaxic classes
  that cannot be easily parsed by a common parser (#496, @octachron).
```

Is this still correct?
Question: Can these "distinct syntaxic (syntactic?) classes" never "be easily parsed..."
OR ... are these specifically classes that can't be easily parsed?

Reply:
Would the following work:
```suggestion
  - Add a `special_function'` variant that directly takes a `Longident.t`
  argument in order to avoid issues with `Longident.t` covering distinct syntaxic classes
  that cannot be easily parsed by a common parser (#496, @octachron).
```

Reply:
Looks good! 

Just double checking that it's "syntaxic" rather than "syntactic."

Reply:
`syntactic` is probably more correct. Another option would be to drop the grammar description:

>  Add `special_function'` variant that directly takes a `Longident.t` argument to avoid ambiguities when parsing generic `Longident.t`

Reply:
Ouh yeay my bad I missed that!